src/Umbraco.Web.BackOffice/Controllers/TourController.cs

```diff
@@ -140,21 +151,71 @@ public async Task<IEnumerable<BackOfficeTourFile>> GetTours()
         return result.Except(toursToBeRemoved).OrderBy(x => x.FileName, StringComparer.InvariantCultureIgnoreCase);
     }
 
-    private async Task<string> GetContentFromEmbeddedResource(string fileName)
+    private IEnumerable<Tuple<IFileInfo, string>> GetTourFiles(IFileProvider fileProvider, string folder)
     {
-        Stream? resourceStream = GetType().Assembly.GetManifestResourceStream(fileName);
+        IEnumerable<IFileInfo> pluginFolders = fileProvider.GetDirectoryContents(folder).Where(x => x.IsDirectory);
 
-        if (resourceStream is null)
+        foreach (IFileInfo pluginFolder in pluginFolders)
         {
-            return string.Empty;
+            var pluginFilters = _filters.Where(x => x.PluginName != null && x.PluginName.IsMatch(pluginFolder.Name)).ToList();
+
+            // If there is any filter applied to match the plugin only (no file or tour alias) then ignore the plugin entirely
+            var isPluginFiltered = pluginFilters.Any(x => x.TourFileName == null && x.TourAlias == null);
+            if (isPluginFiltered)
+            {
+                continue;
+            }
+
+            // get the full virtual path for the plugin folder
+            var pluginFolderPath = WebPath.Combine(folder, pluginFolder.Name);
+
+            // loop through the folder(s) in order to find tours
+            //  - there could be multiple on case sensitive file system
+            // Hard-coding the "backoffice" directory name to gain a better performance when traversing the App_Plugins directory
+            foreach (var subDir in GetToursFolderPaths(fileProvider, pluginFolderPath, "backoffice"))
+            {
+                IEnumerable<IFileInfo> tourFiles = fileProvider
+                    .GetDirectoryContents(subDir)
+                    .Where(x => !string.IsNullOrEmpty(x.PhysicalPath))
+                    .Where(x => x.Name.InvariantEndsWith(".json"));
+
+                foreach (IFileInfo file in tourFiles)
+                {
+                    yield return new Tuple<IFileInfo, string>(file, pluginFolder.Name);
+                }
+            }
         }
+    }
 
-        using var reader = new StreamReader(resourceStream, Encoding.UTF8);
-        return await reader.ReadToEndAsync();
+    private static IEnumerable<string> GetToursFolderPaths(IFileProvider fileProvider, string path, string subDirName)
+    {
+        // Hard-coding the "tours" directory name to gain a better performance when traversing the sub directories
+        var toursDirName = "tours";
```

Comment:
```suggestion
        const string toursDirName = "tours";
```

Suggest making this const 😄